src/rust/engine/process_execution/src/local.rs

```diff
@@ -633,13 +633,17 @@ pub async fn prepare_workdir(
 
   // Capture argv0 as the executable path so that we can test whether we have created it in the
   // sandbox.
-  let maybe_executable_path = RelativePath::new(&req.argv[0]).map(|relative_path| {
-    if let Some(working_directory) = &req.working_directory {
-      working_directory.join(relative_path)
+  let maybe_executable_path = {
+    let mut executable_path = PathBuf::from(&req.argv[0]);
+    if executable_path.is_relative() {
+      if let Some(working_directory) = &req.working_directory {
+        executable_path = working_directory.as_ref().join(executable_path)
```

Comment:
This should probably be using RelativePath so that ../ are followed and collapsed and there is still verification the root has not been escaped. I can't see the harm if it is escaped, but the effort to avoid the lock is nominally being made, so go all the way?

Reply:
Just saw this, sorry. I agree, but I also don't see the harm in letting it escape. Will let it slide for now to save a tree.